modules/generator/processor/localblocks/processor.go

```diff
@@ -547,10 +547,20 @@ func (p *Processor) deleteOldBlocks() (err error) {
 
 		flushedTime := b.FlushedTime()
 		if flushedTime.IsZero() {
+
+			if b.BlockMeta().EndTime.Before(cuttoff) { // Not flushed and old
```

Comment:
I'm not sure we want to add this case.  The original design was to try to flush forever like the ingesters, so that it tolerates network outages and doesn't lose data.   

But mainly this fix isn't addressing the problem we saw.   After enabling "flush_to_storage" this reset the timer on all blocks, because line 553 below is checking based on `flushedTime`. Therefore it keeps blocks for up to 2*complete_block_time.

Reply:
Changed 👍